runtime/bindings/python/vm.cc

```diff
@@ -159,24 +159,24 @@ VmModule VmModule::FromFlatbufferBlob(VmInstance* instance,
   iree_vm_module_t* module = nullptr;
 
   // Bridge to the C-based deallocator API.
-  auto* raw_ptr = flatbuffer_blob.ptr();
+  PyObject* pyobject_ptr = flatbuffer_blob_object.ptr();
   auto ctl_fn = +([](void* self, iree_allocator_command_t command,
                      const void* params, void** inout_ptr) {
     assert(command == IREE_ALLOCATOR_COMMAND_FREE);
-    PyObject* object_ptr = static_cast<PyObject*>(*inout_ptr);
-    Py_XDECREF(object_ptr);
+    PyObject* pyobject_ptr = static_cast<PyObject*>(self);
+    Py_XDECREF(pyobject_ptr);
     return iree_ok_status();
   });
-  flatbuffer_blob.inc_ref();
-  iree_allocator_t deallocator{/*self=*/NULL, /*ctl=*/ctl_fn};
+  Py_XINCREF(pyobject_ptr);
+  iree_allocator_t deallocator{/*self=*/pyobject_ptr, /*ctl=*/ctl_fn};
 
   auto status = iree_vm_bytecode_module_create(
       instance->raw_ptr(),
       {static_cast<const uint8_t*>(buffer_info.ptr),
        static_cast<iree_host_size_t>(buffer_info.size)},
       deallocator, iree_allocator_system(), &module);
   if (!iree_status_is_ok(status)) {
-    iree_allocator_free(deallocator, raw_ptr);
+    iree_allocator_free(deallocator, pyobject_ptr);
```

Comment:
this should be `Py_XDECREF(pyobject_ptr);` to balance the `Py_XINCREF` above
```suggestion
    Py_XDECREF(pyobject_ptr);
```